build.gradle

```diff
@@ -76,16 +112,25 @@ tasks.named('compilePeersJava') {
 tasks.named('compileTestJava') {
     dependsOn(copyLibs)
     dependsOn(copyResources)
+    shouldRunAfter(modifyAndCompileD)
```

Comment:
This dependency should be the other way around: in the declaration of `modifyAndCompileD`, that task should run after `compileTestJava`, so we first compile the tests normally and then modify `D.java` and re-compile it.